Added various features to permit running of tests by others

halindrome opened the pull request:
A tester commented that they need to pass additional parameters to their protocol engine.  Also that their engine limits the scope of targets.  These changes allow for optional specification of those, where the default behavior is as it was earlier.

@BigBlueHat can you verify that this looks okay?  Rob and Hugo have already tested using it.

halindrome committed:
Added handling for query strings and explicit target URIs

halindrome committed:
Added handling for query strings and explicit target URIs

halindrome committed:
Fixed id checking and link header check

halindrome committed:
Fixed id checking and link header check

halindrome committed:
Merge branch 'annotation-protocol-corrections' into annotation-auth

fixes link header check for annotation resources.
fixes dealing with last and first properties that are objects instead of
strings.

These changes were approved by the working group on 11 November.

halindrome committed:
Made tests more tolerant of errors

Added testing of types of returned information.
Divided promise chains to reduce interdependency of tests.
Corrected a false pass caused by nested promises.

halindrome committed:
Made tests more tolerant of errors

Added testing of types of returned information.
Divided promise chains to reduce interdependency of tests.
Corrected a false pass caused by nested promises.

halindrome committed:
Changed annotation content checking to 'type'

halindrome committed:
Changed annotation content checking to 'type'

halindrome commented:
Note that this PR also addresses the annotation content of a 'page'.  We were checking for @context, but that is not required because it cascades.  Changed that to checking for 'type'.

azaroth42 commented:
:+1: Independently verified that the changes fix the known issues with the tests.  Please merge, so we can get back to the implementers and they can resume checking their implementations :)

halindrome committed:
Merge branch 'annotation-auth' of github.com:Spec-Ops/web-platform-tests into annotation-auth

halindrome committed:
Re-introduced URI fixing after rebase.

halindrome committed:
Merge branch 'annotation-auth' of github.com:Spec-Ops/web-platform-tests into annotation-auth

halindrome committed:
Merged in changes from upstream

halindrome commented:
Sorry for the nonsense in this branch.  In order to proceed with testing I needed to rebase so that all the other changes were available in this branch (since this is not yet merged) so that CR testing could proceed.

halindrome committed:
Merged in changes from upstream master.

halindrome committed:
Split out canonical test

The annotation creation test(s) were implicitly testing for preservation
of a canonical URI instead of doing it explicitly.  Split that test out
into a separate one with its own creation so that implementations that
fail the canonical test can still get results on the other parts of
creation.

halindrome commented:
@BigBlueHat and @azaroth42 can you please review again now?  I split out the canonical test as per the request from Europeana.  I have re-run against the RI and CC implementations and updated the test-results at https://w3c.github.io/test-results/annotation-protocol/all.html

halindrome committed:
Changed so the tester can provide their own annotations.

halindrome committed:
Fixed use of If-Match

Added If-Match to update request as per working group on 2 December
2016.  Spec says that updates SHOULD use If-Match.

Corrected Etag to ETag in reference implementation.

halindrome committed:
Added comments for test numbers
